assets/training/model_management/environments/foundation-model-inference/context/score.py

```diff
@@ -585,65 +526,172 @@ def get_processed_input_data_for_chat_completion(data: List[str]) -> str:
+            response_str = client.generate(q, **params).generated_text
+            time_taken = time.time() - time_start
+            logger.info(f"query {i} - time_taken: {time_taken}")
+            results[str(i)] = [f"{response_str}"]
+
+        resp = pd.DataFrame(results)
+        return get_safe_response(resp)
+
+    except Exception as e:
+        return json.dumps({"error": "Error in processing request", "exception": str(e)})
+
+
+if __name__ == "__main__":
+    logger.info(init())
+    assert task_type is not None
+
+    valid_inputs = {
+        "text-generation": [
+            {
+                "input_data": {
+                    "input_string": ["the meaning of life is"],
+                    "parameters": {"max_new_tokens": 100, "do_sample": True},
+                }
+            }
+        ],
+        "chat-completion": [
+            {
+                "input_data": {
+                    "input_string": [
+                        {
+                            "role": "user",
+                            "content": "What is the tallest building in the world?",
+                        },
+                        {
+                            "role": "assistant",
+                            "content": "As of 2021, the Burj Khalifa in Dubai, United Arab Emirates is the tallest building in the world, standing at a height of 828 meters (2,722 feet). It was completed in 2010 and has 163 floors. The Burj Khalifa is not only the tallest building in the world but also holds several other records, such as the highest occupied floor, highest outdoor observation deck, elevator with the longest travel distance, and the tallest freestanding structure in the world.",
+                        },
+                        {"role": "user", "content": "and in Africa?"},
+                        {
+                            "role": "assistant",
+                            "content": "In Africa, the tallest building is the Carlton Centre, located in Johannesburg, South Africa. It stands at a height of 50 floors and 223 meters (730 feet). The CarltonDefault Centre was completed in 1973 and was the tallest building in Africa for many years until the construction of the Leonardo, a 55-story skyscraper in Sandton, Johannesburg, which was completed in 2019 and stands at a height of 230 meters (755 feet). Other notable tall buildings in Africa include the Ponte City Apartments in Johannesburg, the John Hancock Center in Lagos, Nigeria, and the Alpha II Building in Abidjan, Ivory Coast",
+                        },
+                        {"role": "user", "content": "and in Europe?"},
+                    ],
+                    "parameters": {
+                        "temperature": 0.9,
+                        "top_p": 0.6,
+                        "do_sample": True,
+                        "max_new_tokens": 100,
+                    },
+                }
+            }
+        ],
+    }
+
+    for sample_ip in valid_inputs[task_type]:
```

Comment:
do we want to run these?

Reply:
I don't think main is called? (its just there for debugging afaik)

Reply:
(removed it anyway)